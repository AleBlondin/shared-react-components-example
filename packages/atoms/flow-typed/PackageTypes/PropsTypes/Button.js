// @flow

import React from 'react';

declare type ButtonPropsType = {|
  children: React.Node,
  className?: string,
  disabled?: boolean,
  light?: boolean,
  onClick?: (event: EventType) => void,
  type?: string,
  small?: boolean,
|};
