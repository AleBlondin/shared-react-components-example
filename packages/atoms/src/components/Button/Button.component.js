// @flow

import * as React from 'react';

const Button = (props: ButtonPropsType) => (
  <button
    disabled={props.disabled}
    onClick={props.onClick}
    type={props.type}
    className={props.className}
  >
    {props.children}
  </button>
);

Button.defaultProps = {
  disabled: false,
  children: null,
};
Button.displayName = 'Button';

export default Button;
