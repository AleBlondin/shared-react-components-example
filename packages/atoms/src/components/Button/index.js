// @flow

export { default } from './Button.style';
