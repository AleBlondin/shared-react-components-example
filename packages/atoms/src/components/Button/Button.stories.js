import React from 'react';

// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { boolean, withKnobs } from '@storybook/addon-knobs';

import Button from './Button.style';

storiesOf('Atoms', module)
  .addDecorator(withKnobs)
  .add('Button', () => (
    <Button
      disabled={boolean('disabled', false)}
      small={boolean('small', false)}
      light={boolean('light', false)}
    >
      <span className="button-label">label</span>
    </Button>
  ));
