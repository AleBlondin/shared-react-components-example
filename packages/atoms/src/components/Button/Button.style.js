// @flow

import styled from 'styled-components';

import {
  SharedComponentsColors,
  typography,
  Grid,
} from 'shared-components-atoms/style';

import Button from './Button.component';

type StyledPropsType = {
  disabled: boolean,
  light: boolean,
  small: boolean,
};

export default styled(Button)`
  ${({ light, disabled, small }: StyledPropsType) => `
    border: 1px solid ${SharedComponentsColors.brandGreen};
    border-radius: 3px;
    ${disabled ? '' : 'cursor: pointer;'}
    height: ${small ? '18px' : Grid(10)};
    line-height: 17px;
    ${
      light
        ? `
      ${typography.lightText}
      background-color: transparent;
      color: ${SharedComponentsColors.brandGreen};
    `
        : `${typography.normalText}
      background-color: ${SharedComponentsColors.brandGreen};
      color: ${SharedComponentsColors.white};
    `
    }
    ${
      small
        ? `
        font-weight: 300;
        font-size: 12px
        `
        : `width: 100%`
    }
    :disabled {
      background: #8ED0AF;
      border-color: #8ED0AF;
    }`};
`;
