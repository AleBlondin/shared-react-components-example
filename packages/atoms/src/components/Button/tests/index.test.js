import React from 'react';
import renderer from 'react-test-renderer';

import Button from '..';

describe('<Button />', () => {
  it('should match snapshot for normal button', () => {
    const props = {
      textLabel: 'AUTH_CONNEXION',
      type: 'button',
      className: 'button',
      onClick: () => null,
    };
    const renderedComponent = renderer.create(<Button {...props} />);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should match snapshot for light button', () => {
    const props = {
      textLabel: 'AUTH_CONNEXION',
      type: 'button',
      className: 'button',
      onClick: () => null,
      light: true,
    };
    const renderedComponent = renderer.create(<Button {...props} />);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should match snapshot for small button', () => {
    const props = {
      textLabel: 'AUTH_CONNEXION',
      type: 'button',
      className: 'button',
      onClick: () => null,
      small: true,
    };
    const renderedComponent = renderer.create(<Button {...props} />);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should match snapshot for gradient button', () => {
    const props = {
      textLabel: 'AUTH_CONNEXION',
      type: 'button',
      className: 'button',
      onClick: () => null,
      gradient: {
        color: '#AAAAAA',
        finalColor: '#FFFFFF',
        initialColor: '#000000',
      },
    };
    const renderedComponent = renderer.create(<Button {...props} />);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
    expect(renderedComponent).toMatchSnapshot();
  });
});
