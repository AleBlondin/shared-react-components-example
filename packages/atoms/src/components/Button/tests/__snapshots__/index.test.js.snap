// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot for gradient button 1`] = `
.c0 {
  border: 1px solid #00965E;
  border-radius: 3px;
  cursor: pointer;
  height: 50px;
  line-height: 17px;
  color: #626262;
  font-family: "Lato";
  font-size: 14px;
  font-weight: 400;
  background-color: #00965E;
  color: #FFFFFF;
  width: 100%;
}

.c0:disabled {
  background: #8ED0AF;
  border-color: #8ED0AF;
}

<button
  className="button c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> should match snapshot for gradient button 2`] = `
.c0 {
  border: 1px solid #00965E;
  border-radius: 3px;
  cursor: pointer;
  height: 50px;
  line-height: 17px;
  color: #626262;
  font-family: "Lato";
  font-size: 14px;
  font-weight: 400;
  background-color: #00965E;
  color: #FFFFFF;
  width: 100%;
}

.c0:disabled {
  background: #8ED0AF;
  border-color: #8ED0AF;
}

<button
  className="button c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> should match snapshot for light button 1`] = `
.c0 {
  border: 1px solid #00965E;
  border-radius: 3px;
  cursor: pointer;
  height: 50px;
  line-height: 17px;
  color: #626262;
  font-family: "Lato";
  font-size: 14px;
  font-weight: 300;
  background-color: transparent;
  color: #00965E;
  width: 100%;
}

.c0:disabled {
  background: #8ED0AF;
  border-color: #8ED0AF;
}

<button
  className="button c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> should match snapshot for light button 2`] = `
.c0 {
  border: 1px solid #00965E;
  border-radius: 3px;
  cursor: pointer;
  height: 50px;
  line-height: 17px;
  color: #626262;
  font-family: "Lato";
  font-size: 14px;
  font-weight: 300;
  background-color: transparent;
  color: #00965E;
  width: 100%;
}

.c0:disabled {
  background: #8ED0AF;
  border-color: #8ED0AF;
}

<button
  className="button c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> should match snapshot for normal button 1`] = `
.c0 {
  border: 1px solid #00965E;
  border-radius: 3px;
  cursor: pointer;
  height: 50px;
  line-height: 17px;
  color: #626262;
  font-family: "Lato";
  font-size: 14px;
  font-weight: 400;
  background-color: #00965E;
  color: #FFFFFF;
  width: 100%;
}

.c0:disabled {
  background: #8ED0AF;
  border-color: #8ED0AF;
}

<button
  className="button c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> should match snapshot for normal button 2`] = `
.c0 {
  border: 1px solid #00965E;
  border-radius: 3px;
  cursor: pointer;
  height: 50px;
  line-height: 17px;
  color: #626262;
  font-family: "Lato";
  font-size: 14px;
  font-weight: 400;
  background-color: #00965E;
  color: #FFFFFF;
  width: 100%;
}

.c0:disabled {
  background: #8ED0AF;
  border-color: #8ED0AF;
}

<button
  className="button c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> should match snapshot for small button 1`] = `
.c0 {
  border: 1px solid #00965E;
  border-radius: 3px;
  cursor: pointer;
  height: 18px;
  line-height: 17px;
  color: #626262;
  font-family: "Lato";
  font-size: 14px;
  font-weight: 400;
  background-color: #00965E;
  color: #FFFFFF;
  font-weight: 300;
  font-size: 12px;
}

.c0:disabled {
  background: #8ED0AF;
  border-color: #8ED0AF;
}

<button
  className="button c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> should match snapshot for small button 2`] = `
.c0 {
  border: 1px solid #00965E;
  border-radius: 3px;
  cursor: pointer;
  height: 18px;
  line-height: 17px;
  color: #626262;
  font-family: "Lato";
  font-size: 14px;
  font-weight: 400;
  background-color: #00965E;
  color: #FFFFFF;
  font-weight: 300;
  font-size: 12px;
}

.c0:disabled {
  background: #8ED0AF;
  border-color: #8ED0AF;
}

<button
  className="button c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;
